tests/test_args_specs.py

```diff
@@ -311,6 +325,43 @@ def validate_permissions_value(self, permissions):
             # should only have a letter once
             valid_letters.remove(letter)
 
+    def test_model_specs_serialize(self):
+        """ARGS_SPEC: test each ARGS_SPEC can serialize to JSON."""
+        from natcap.invest import spec_utils
+
+        for model_name, metadata in cli._MODEL_UIS.items():
+            model = importlib.import_module(metadata.pyname)
+            try:
+                _ = spec_utils.serialize_args_spec(model.ARGS_SPEC)
+            except TypeError:
+                self.fail(
```

Comment:
Neat, I haven't seen `self.fail` before! Could we also show the original error traceback?

Reply:
Yeah `self.fail` is new to me too. This seems like the preferred pattern to assert that a specific exception is avoided.

Good idea adding the original error. It's hard (but not impossible) to get the full traceback without ever raising the error. And this `TypeError` should always be the one explicitly raised by `spec_utils.fallback_serializer` so I think it's okay to just include that message without traceback. An example with these changes:

```
E               AssertionError: Failed to avoid TypeError when serializing natcap.invest.carbon.ARGS_SPEC:
E               fallback serializer is missing for <class 'module'>
```